File: app/src/main/java/com/systemvx/androiddevtest/OrderComplaintAdapter.java
package com.systemvx.androiddevtest.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.core.content.ContextCompat;
import androidx.recyclerview.widget.RecyclerView;

import com.systemvx.androiddevtest.R;
import com.systemvx.androiddevtest.bean.OrderTypeBean;

import java.util.List;

public class OrderComplaintAdapter extends RecyclerView.Adapter {
    private List<OrderTypeBean> list;
    private Context context;
    public OnItemClick onItemClick;

    public interface OnItemClick{
        void OnItemClick(View view,int position);
    }
    public OrderComplaintAdapter(Context context,List<OrderTypeBean> list) {
        this.list = list;
        this.context = context;
    }
    public void setOnItemClickLisener(OnItemClick clickLisener){
        onItemClick = clickLisener;
    }
    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.item_compaint_type, null, false);
        return new MyViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int position) {
        MyViewHolder viewHolder = (MyViewHolder) holder;
        viewHolder.itemView.setOnClickListener(view -> onItemClick.OnItemClick(view,position));

        if (list.get(position).isCheck()){
            viewHolder.itemView.setBackgroundColor(ContextCompat.getColor(context,R.color.purple_200));
        }else {
            viewHolder.itemView.setBackgroundColor(ContextCompat.getColor(context,R.color.white));
        }
    }

    @Override
    public int getItemCount() {
        return list.size();
    }
    class MyViewHolder extends RecyclerView.ViewHolder{

        public MyViewHolder(@NonNull View itemView) {
            super(itemView);
        }
    }
}

File: app/src/main/java/com/systemvx/androiddevtest/ui/complaint/OrderComplaintAdapter.kt
package com.systemvx.androiddevtest.ui.complaint

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.RecyclerView
import com.systemvx.androiddevtest.R
import com.systemvx.androiddevtest.data.OrderTypeBean

class OrderComplaintAdapter(private val context: Context, private val list: List<OrderTypeBean>) :
        RecyclerView.Adapter<OrderComplaintAdapter.MyViewHolder>() {
    var onItemClick: OnItemClick? = null

    interface OnItemClick {
        fun OnItemClick(view: View?, position: Int)
    }

    fun setOnItemClickLisener(clickLisener: OnItemClick?) {
        onItemClick = clickLisener
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val view = LayoutInflater.from(context).inflate(R.layout.item_compaint_type, null, false)
        return MyViewHolder(view)
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        holder.itemView.setOnClickListener { view: View? -> onItemClick!!.OnItemClick(view, position) }
        if (list[position].isCheck) {
            holder.itemView.setBackgroundColor(ContextCompat.getColor(context, R.color.purple_200))
        } else {
            holder.itemView.setBackgroundColor(ContextCompat.getColor(context, R.color.white))
        }
    }

    override fun getItemCount(): Int {
        return list.size
    }

    class MyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView)
}